improver/developer_tools/metadata_interpreter.py

```diff
@@ -0,0 +1,563 @@
+                                point == bounds
+                                for point, bounds in zip(
+                                    cube.coord("time").points, upper_bounds
+                                )
+                            ]
+                        ):
+                            self.errors.append(
+                                "Time points should be equal to upper bounds"
+                            )
+
+            elif cm in NONCOMP_CMS or cm.method in NONCOMP_CM_METHODS:
+                self.errors.append(f"Non-standard cell method {cm}")
+            else:
+                # flag method which might be invalid, but we can't be sure
+                self.warnings.append(
+                    f"Unexpected cell method {cm}. Please check the standard to "
+                    "ensure this is valid"
+                )
+
+    def _check_blend_and_model_attributes(self, attrs):
+        """Interprets attributes for model and blending information
+        and checks for self-consistency"""
+        self.blended = True if BLEND_TITLE_SUBSTR in attrs["title"] else False
+
+        if self.blended:
+            if self.model_id_attr not in attrs:
+                self.errors.append(f"No {self.model_id_attr} on blended file")
+            else:
+                codes = attrs[self.model_id_attr].split(" ")
+                names = [MODEL_NAMES[code] for code in codes]
+                self.model = ", ".join(names)
+
+        else:
+            if self.model_id_attr in attrs:
+                for key in MODEL_CODES:
+                    if (
+                        f"{key} Model" in attrs["title"]
+                        and attrs[self.model_id_attr] != MODEL_CODES[key]
+                    ):
+                        self.errors.append(
+                            f"Title {attrs['title']} is inconsistent with model ID "
+                            f"attribute {attrs[self.model_id_attr]}"
+                        )
+
+                try:
+                    self.model = MODEL_NAMES[attrs[self.model_id_attr]]
+                except KeyError:
+                    self.errors.append(
+                        f"Attribute {attrs[self.model_id_attr]} is not a valid single "
+                        "model.  If valid for blend, then title attribute is missing "
+                        f"expected substring {BLEND_TITLE_SUBSTR}."
+                    )
+
+    def check_attributes(self, attrs):
+        """Checks for unexpected attributes, then interprets values for model
+        information and checks for self-consistency"""
+        try:
+            permitted_attributes = COMP_ATTRS + DIAG_ATTRS[self.diagnostic]
+        except KeyError:
+            permitted_attributes = COMP_ATTRS.copy()
+
+        if any([attr in NONCOMP_ATTRS for attr in attrs]):
+            self.errors.append(
+                f"Attributes {attrs.keys()} include one or more forbidden "
+                f"values {NONCOMP_ATTRS}"
+            )
```

Comment:
This would list only the matching forbidden attributes, rather than all of them.
```suggestion
        if any([attr in NONCOMP_ATTRS for attr in attrs]):
            self.errors.append(
                f"Attributes {attrs.keys()} include forbidden "
                f"values {[attr for attr in attrs if attr in NONCOMP_ATTRS]}"
            )
```

Reply:
Elegant :+1: